warehouse/upload.go

```diff
@@ -1153,6 +1184,58 @@ func (job *UploadJobT) createLoadFiles(generateAll bool) (startLoadFileID int64,
 	return startLoadFileID, endLoadFileID, nil
 }
 
+func (job *UploadJobT) setStagingFileSuccess(stagingFileIDs []int64) {
+	sqlStatement := fmt.Sprintf(`UPDATE %s SET status=$1, updated_at=$2 WHERE id=ANY($3)`, warehouseutils.WarehouseStagingFilesTable)
```

Comment:
Minor comment, ignore if not needed. Should we use `where id IN $3` instead of `where id=ANY($3)`

Both should work, but `IN` seems to be more intuitive and we use that across other modules.

Reply:
ANY seems to be more verstaile https://stackoverflow.com/a/34627688
Also pq.Array can be used instead of us preparing a comma separated list of values

Reply:
I would suggest that we start using ANY here and switch everywhere else with IN. It seems to be a better way to construct queries from code.

Reply:
https://stackoverflow.com/a/34627688 ->

This says IN uses indexes and ANY doesn't.

Reply:
I too was confused when I saw that and had done an EXPLAIN on both variants in our query like so
`EXPLAIN ANALYZE UPDATE wh_staging_files SET status='success', updated_at='2021-01-01' WHERE id=ANY('{3,2,1}');
`
`EXPLAIN ANALYZE UPDATE wh_staging_files SET status='success', updated_at='2021-01-01' WHERE id IN (3,2,1);`

Both of them had given the same query plan. I had then read the reply again and what you mentioned along with the other link below it are special cases where the index wasn't used and we don't really use IN or ANY in that way afaik.

The one that you saw, points to this link -> https://dba.stackexchange.com/questions/125413/index-not-used-with-any-but-used-with-in/125500#125500 which is the special case the answer discusses.

Reply:
Just checked this. 

For composite indexes, ANY is not using the index but IN is using it. Since this is a primary key, we will go ahead with ANY.